R/stubs.R

```diff
@@ -24,5 +24,73 @@ sourceMarkers <- function(name, markers, basePath = NULL,
 
 #' @export
 navigateToFile <- function(file, line = 1L, column = 1L) {
-  callFun("navigateToFile", file, as.integer(line), as.integer(column))  
+  callFun("navigateToFile", file, as.integer(line), as.integer(column))
+}
+
+#' Replace Text within a Document
+#'
+#' Use this to change the contents of an open document.
+#'
+#' @param ranges A list of ranges. Each range is a four-element
+#'   integer vector, defining a document range with coordinates:
+#'
+#'   \code{startRow, startColumn, endRow, endColumn}
+#'
+#' @param text A character vector, indicating what text should be
+#'   inserted at each aforementioned range. This should either
+#'   be length one (in which case, this text is applied to each
+#'   range specified); otherwise, it should be the same length
+#'   as the \code{ranges} list.
+#'
+#' @param id The document id. When \code{NULL} or blank,
+#'   the mutation will apply to the currently open, or last
+#'   focused, RStudio document. Use the \code{id} returned
+#'   from \code{\link{getActiveDocumentContext}()} to ensure
+#'   that the operation is applied on the intended document.
+#'
+#' @export
+#' @family Source Document Methods
+replaceRanges <- function(ranges, text, id = "") {
+  callFun("replaceRanges", ranges, text, id)
+}
+
+#' Replace Selection within a Document
+#'
+#' Change the selection in a document.
```

Comment:
This is ambiguous - does it modify what text is selected, or the contents of the selection? (It's obvious from the arguments, but it would be good to be a little clearer here)